nixos/modules/services/hardware/asusd.nix

```diff
@@ -87,18 +101,20 @@ in
         maybeConfig =
           name: cfg:
           lib.mkIf (cfg != null) {
-            source = pkgs.writeText name cfg;
+            source = if builtins.isPath cfg || lib.isStorePath cfg then cfg else pkgs.writeText name cfg;
```

Comment:
I really dislike this. Maybe we should introduce `text` and `source` suboptions, similar to what `environment.etc` itself does?